Tools/WrapperGenerator/CSharpWrapperGenerator.py

```diff
@@ -188,6 +213,26 @@ def __get_file_output_directory(
 
         return "/".join(out_path)
 
+    def __render_named_object(self, named_object, enum_template) -> None:
+        """Given an object with a "name" attribute, render it to the corresponding .cs file."""
+        subdir = self.__get_file_output_directory(named_object)
+        if named_object.surrounding_types:
+            for st in named_object.surrounding_types:
+                subdir = f"{subdir}/{st}"
+
+        Path(self.__OUTPUT_DIRECTORY + subdir).mkdir(parents=True, exist_ok=True)
+
+        file_path = f"{self.__OUTPUT_DIRECTORY}{subdir}/{named_object.name}.cs"
+        with open(file_path, "w", encoding="utf-8") as f:
```

Comment:
Just wanted to check if memory server correctly but the action with open allow for automatic file close on scope exit?

Reply:
That's correct.